src/shell/shell_misc.cpp

```diff
@@ -166,10 +166,17 @@ void DOS_Shell::InputCommand(char * line) {
 					break;
 				case 0x53:/* DELETE */
 					{
-						if(str_index>=str_len) break;
-						Bit16u a=str_len-str_index-1;
-						Bit8u* text=reinterpret_cast<Bit8u*>(&line[str_index+1]);
-						DOS_WriteFile(STDOUT,text,&a);//write buffer to screen
+						if (str_index >= str_len)
+							break;
+
+						// Confirm that the tail length is in-bounds before casting
+						const auto tail_length = str_len - str_index - 1;
+						assert(tail_length >= 0 && tail_length <= UINT16_MAX);
+
+						// Once confirmed, cast into the required type
+						auto amount = static_cast<uint16_t>(tail_length);
+						auto text = reinterpret_cast<uint8_t *>(&line[str_index + 1]);
+						DOS_WriteFile(STDOUT,text, &amount);//write buffer to screen
```

Comment:
Ehh, the real problem in here is badly designed interface to `DOS_WriteFile`…

This change is ok, except I strongly dislike variable name `amount`, as number of bytes can be counted :)